libfwupdplugin/fu-context.c

```diff
@@ -786,21 +789,44 @@ fu_context_load_hwinfo(FuContext *self, FuContextHwidFlags flags, GError **error
 	g_return_val_if_fail(FU_IS_CONTEXT(self), FALSE);
 	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
 
+	if ((flags & FU_CONTEXT_HWID_FLAG_LOAD_CONFIG) > 0) {
+		g_autoptr(GError) error_local = NULL;
+		if (!fu_hwids_config_setup(self, priv->hwids, &error_local)) {
+			if (!g_error_matches(error_local, FWUPD_ERROR, FWUPD_ERROR_NOT_SUPPORTED))
+				g_warning("Failed to load HWIDs config: %s", error_local->message);
+		}
+	}
+	if ((flags & FU_CONTEXT_HWID_FLAG_LOAD_DMI) > 0) {
+		g_autoptr(GError) error_local = NULL;
+		if (!fu_hwids_dmi_setup(self, priv->hwids, &error_local)) {
+			if (!g_error_matches(error_local, FWUPD_ERROR, FWUPD_ERROR_NOT_SUPPORTED))
+				g_warning("Failed to load HWIDs DMI: %s", error_local->message);
+		}
+	}
+	if ((flags & FU_CONTEXT_HWID_FLAG_LOAD_FDT) > 0) {
+		g_autoptr(GError) error_local = NULL;
+		if (!fu_hwids_fdt_setup(self, priv->hwids, &error_local)) {
+			if (!g_error_matches(error_local, FWUPD_ERROR, FWUPD_ERROR_NOT_SUPPORTED))
+				g_warning("Failed to load HWIDs FDT: %s", error_local->message);
+		}
+	}
+	if ((flags & FU_CONTEXT_HWID_FLAG_LOAD_KENV) > 0) {
+		g_autoptr(GError) error_local = NULL;
+		if (!fu_hwids_kenv_setup(self, priv->hwids, &error_local)) {
+			if (!g_error_matches(error_local, FWUPD_ERROR, FWUPD_ERROR_NOT_SUPPORTED))
+				g_warning("Failed to load HWIDs kenv: %s", error_local->message);
+		}
+	}
```

Comment:
If the failure is not not supported, shouldn't this be more critical?  Like propagate the error up?

Reply:
Good idea. Fixed.